### DISCUSSION TITLE
Is it safe to use the CacheProvider in Prod, or only enable it in dev mode?

### DISCUSSION THREAD

**Discussion by Vxee** [emotion-js/emotion#3205] [Q&A]
Now our project is using MUI & emotion, I found some pages will dynamically insert 5000+ style tags in development, the page performance is not very good. I find some solutions such as use @emotion/cache and @emotion/babel-plugin. But I don't know is there any I need to care when used in PROD. 

Because I find in PROD, the style tag will be combined, seems like will not have performance issue. So maybe don't need to use CacheProvider.

**Comment by Andarist** [comment]
I'd recommend trying to wrap both environments with a CacheProvider that would provide a speedy cache. If you still performance problems then I'd investigate if you are not generating too many styles of too many dynamic rules. If you have animations that set new values in styles on each frame that's a recipe for a disaster.

**Reply by Vxee** [reply]
I try to wrap CacheProvider in DEV, but the style tags count doesn't decrease. Does this meet expectations?
